### ISSUE TITLE
Apollo client not working due to "Invalid response for blob" 

### ISSUE THREAD

**Issue by gtRfnkN** [microsoft/react-native-windows#12187]
### Problem Description

There seems to be an issue using the `@apollo/client` library. When that library is added, running the app without the web debugger results in errors whenever `fetch()` is called or something from the `@apollo/client` is imported. Everything works fine via the web debugger and also after removing the apollo client library.

The issue might be related to https://github.com/microsoft/react-native-windows/issues/11439 but should be resolved in the latest RNW versions. I've tested this with both RNW 0.71.38 & 0.72.9 and got the same errors. We've upgraded from version 0.69.6, where everything worked fine.

### Steps To Reproduce

1. set up a new app following the RNW "Get Started" guide
2. add `@apollo/client` and `graphql` dependencies following their guides
3. create a simple apollo wrapper following their guides
4. open the app and disable the web debugger
5. the app shows a blank page and there are multiple errors in the console:
`Error: Invalid response for blob` and `Error: ENOENT: no such file or directory, open '\\?\E:\git\rnw-apollo\code'`

### Expected Results

The app loads normally, renders the UI, and can fetch data

### CLI version

11.3.7

### Environment

```markdown
System:
  OS: Windows 10 10.0.19045
  CPU: "(12) x64 AMD Ryzen 5 3600 6-Core Processor              "
  Memory: 14.48 GB / 31.95 GB
Binaries:
  Node:
    version: 20.7.0 (-> also tested with v18 & v16)
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 1.22.5
    path: C:\Program Files (x86)\Yarn\bin\yarn.CMD
  npm:
    version: 10.1.0
    path: C:\Program Files\nodejs\npm.CMD
  Watchman:
    version: 20200816.222424.0
    path: C:\Users\XXX\AppData\Local\watchman\watchman.EXE
SDKs:
  Android SDK: Not Found
  Windows SDK:
    AllowDevelopmentWithoutDevLicense: Enabled
    AllowAllTrustedApps: Enabled
    Versions:
      - 10.0.14393.0
      - 10.0.17134.0
      - 10.0.17763.0
      - 10.0.18362.0
      - 10.0.19041.0
      - 10.0.22621.0
IDEs:
  Android Studio: Not Found
  Visual Studio:
    - 17.7.34031.279 (Visual Studio Community 2022)
    - 16.11.32126.315 (Visual Studio Community 2019)
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.5
    wanted: 0.72.5
  react-native-windows:
    installed: 0.72.9
    wanted: 0.72.9
npmGlobalPackages:
  "*react-native*": Not Found
```

### Target Platform Version

None

### Target Device(s)

Desktop

### Visual Studio Version

None

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

Here is a basic repository to reproduce the error: https://github.com/gtRfnkN/rnw-apollo

Start the app via `yarn windows` . The app should start normally and show 2 buttons: one to load an apollo component and another to trigger a `fetch()`. Clicking either button throws errors.

**Comment by chrisglein** [comment]
@JunielKatarn They're on a version that should have the #11439 fix. Is there something else required here?

**Comment by gtRfnkN** [comment]
Good morning and thank you for the response.

I have an update: enabling Hermes improves matters and both `fetch()` and apollo's `useQuery` work. Both get their respective data and can be re-run as many times as you like. However, after clicking "Reload Javascript" via the develop menu, both of them throw the same error as before. Restarting only the app (while metro is still running) "fixes" the issue and fetching works again until the next reload.

I've update the linked repository with the mentioned changes: https://github.com/gtRfnkN/rnw-apollo.
You can reproduce it by running `yarn windows` first to deploy the app and see it in a working state. Then "reload javascript" to see the error. And then close and restart the app (with metro still running) to see it working again.

**Comment by Elabar** [comment]
Just run into the same problem. I think it has nothing to do with the library `@apollo/client`. I tried initializing a fresh new project.

Disabling web debugger in `App.cpp`
```diff
#if BUNDLE
    JavaScriptBundleFile(L"index.windows");
    InstanceSettings().UseWebDebugger(false);
    InstanceSettings().UseFastRefresh(false);
#else
    JavaScriptBundleFile(L"index");
-   InstanceSettings().UseWebDebugger(true);
+   InstanceSettings().UseWebDebugger(false);
    InstanceSettings().UseFastRefresh(true);
#endif
```


Run `npx react-native run-windows`, try to click any link in the "Learn More" section, it should work just fine.
Then, reload the app and click the same thing again, you will see the error being thrown.

Let me know if a repro repo is needed.

Environment:
```
react-native: 0.72.6,
react-native-windows: 0.72.12
```

**Comment by gtRfnkN** [comment]
You're right, @Elabar. I've tested a fresh project as you've described and can confirm that the issue occurs there as well. So it's not related to `@apollo/client` - it was just easier to see there because the error was thrown as soon as the `ApolloContext` was created.

That would mean that reloading the app without the web debugger doesn't work with at least RNW 0.72

@chrisglein @JunielKatarn is there something more we can test or provide you with to get this resolved?

**Comment by JunielKatarn** [comment]
@gtRfnkN Reviewing this issue.

We have a sample app to reproduce this issue:
https://github.com/jurocha-ms/Repro/tree/main/ReactNative/fetchwinvalidblob
The bug is expected to happen in RNW version `0.72.2`.
I'll verify whether `0.72.9` is still affected and whether the latest available 0.72 version solves it for your app.

**Comment by JunielKatarn** [comment]
> Just run into the same problem. I think it has nothing to do with the library `@apollo/client`. I tried initializing a fresh new project.
> 
> Disabling web debugger in `App.cpp`
> 
> ```diff
> #if BUNDLE
>     JavaScriptBundleFile(L"index.windows");
>     InstanceSettings().UseWebDebugger(false);
>     InstanceSettings().UseFastRefresh(false);
> #else
>     JavaScriptBundleFile(L"index");
> -   InstanceSettings().UseWebDebugger(true);
> +   InstanceSettings().UseWebDebugger(false);
>     InstanceSettings().UseFastRefresh(true);
> #endif
> ```
> 
> Run `npx react-native run-windows`, try to click any link in the "Learn More" section, it should work just fine. Then, reload the app and click the same thing again, you will see the error being thrown.
> 
> Let me know if a repro repo is needed.
> 
> Environment:
> 
> ```
> react-native: 0.72.6,
> react-native-windows: 0.72.12
> ```

This looks very similar to the triggering method for our sample app.
See https://github.com/jurocha-ms/Repro/blob/main/ReactNative/fetchwinvalidblob/windows/fetchwinvalidblob/App.cpp#L25

Enabling direct debugging and disabling web debugging exposes the bug.
It's possible the only relevant property for this behavior is `UseWebDebugger`.
If this is the same bug (race condition when initializing and running the HTTP and Blob modules), bumping RNW to the latest version in branch 0.72 should suffice.

Will confirm shortly.

**Comment by JunielKatarn** [comment]
I am able to reproduce the issue by loading the dev menu (`CTRL+SHIFT+D`) and reloading the instance.
We recently encountered this behavior internally.
This is a corner case which was not addressed by #11439.

As suspected by some of you, it has nothing to do with the Node dependencies (`@apollo/client`, `graphql`).

<img width="1200" alt="image" src="https://github.com/microsoft/react-native-windows/assets/4507319/d324f8ba-cbcb-404e-aeca-8e59af5caab1">
